portal-web/test/functional/com/liferay/portalweb/macros/LiferayEvent.macro

```diff
@@ -0,0 +1,105 @@
+definition {
+
+	macro initializeLiferayEventLog {
+		var javascript = '''
+const TRACKED_EVENTS = [
+	'addPortlet',
+	'allPortletsReady',
+	'allUploadsComplete',
+	'beforeNavigate',
+	'beforeScreenFlip',
+	'closePortlet',
+	'dataLayoutConfigReady',
+	'destroyPortlet',
+	'editorAPIReady',
+	'endNavigate',
+	'form:registered',
+	'formNavigator:reveal',
+	'dropdownShow',
+	'entitySelectionRemoved',
+	'exitNavigation',
+	'hideNavigationMenu',
+	'initLayout',
+	'inputLocalized:localeChanged',
+	'modalIframeLoaded',
+	'noticeShow',
+	'noticeHide',
+	'popupReady',
+	'portletDragStart',
+	'portletMoved',
+	'portletReady',
+	'ratings:vote',
+	'screenActivate',
+	'screenDeactivate',
+	'screenFlip',
+	'screenLoad',
+	'search-container:registered',
+	'sessionExpired',
+	'showNavigationMenu',
+	'spaRequestTimeout',
+	'startNavigate',
+	'submitForm',
+	'tempFileRemoved',
+	'toggleControls',
+	'updatedLayout',
+];
+
+let __LFR_TRACKED_EVENTS__ = {};
+
+Liferay.on(TRACKED_EVENTS, (event) => {
+	const eventLog = __LFR_TRACKED_EVENTS__[event.type]
+		? [...__LFR_TRACKED_EVENTS__[event.type], event]
+		: [event];
+
+	__LFR_TRACKED_EVENTS__ = {
+		...__LFR_TRACKED_EVENTS__,
+		[event.type]: eventLog,
+	};
```

Comment:
Seems to be no point in copying the `__LFR_TRACKED_EVENTS__` object here.

ie. all of this could be:

```js
const __LFR_TRACKED_EVENTS__ = {};

Liferay.on(TRACKED_EVENTS, (event) =>  {
    const events = __LFR_TRACKED_EVENTS__[event.type] || [];

    __LFR_TRACKED_EVENTS__[event.type] = [...events, event];
});

Reply:
That is correct 👍